### ISSUE TITLE
AtomArrayStack with varying annotations per model

### ISSUE THREAD

**Issue by entropybit** [biotite-dev/biotite#489]
Based on comment from @padix-key on #487 I opened this issue. The problem is that currently it is not possible to manage a collection of models with different annotations on a per model basis via AtomArrayStacks.  Features like occupancy or b_factors and even charges could however vary for different models of even the same molecule. 
This is a conceptual problem with the current layout of AtomArrayStack. 
@padix-key correct me if any of this is wrong please.

**Comment by padix-key** [comment]
Yes, this is correct, currently the only annotation, that may differ between models is `coord`. Probably the most work needs to go into refactoring the file interfaces, the necessary changes to `_AtomArrayBase` should be quite small.

**Comment by francois-rozet** [comment]
Another issue with `AtomArrayStack` is that it does not support models with varying length. It is therefore impossible to read or write files with several models of different length with `biotite`, even though it is perfectly valid in the PDB and PDBx formats.

This could be resolved by making `AtomArrayStack` a list of `AtomArray`, but I don't know what this would break.

**Comment by padix-key** [comment]
Implementing `AtomArrayStack` as a list of `AtomArray` would remove he vectorization over the models making multi-model analysis probably substantially slower.

However we could support also a list[AtomArray] in the `File` classes. This is already tracked in https://github.com/biotite-dev/biotite/issues/131, although this issue is quite stale to be honest.

**Comment by francois-rozet** [comment]
To keep vectorization, you could also concatenate "models" as a single `AtomArray` with a `model_id` column.

**Comment by padix-key** [comment]
This way we would lose the ability to perform operations over models. For example to get the displacement vector of each atom and model compared to the first model, we now can simply run `stack.coord - stack.coord[0]`. This would not be possible anymore if we concatenated every model into the `atom` dimension of an `AtomArray`.
